sklearn/manifold/t_sne.py

```diff
@@ -692,9 +700,11 @@ def _fit(self, X, skip_num_points=0):
 
                 if self.metric == "euclidean":
                     distances = pairwise_distances(X, metric=self.metric,
-                                                   squared=True)
+                                                   squared=True,
+                                                   n_jobs=self.n_jobs)
```

Comment:
for euclidean metric, having n_jobs > 1 unless blas num threads has been limited.
Could you rerun your benchmark with this n_jobs removed ?

Reply:
Yes, of course you are right, somehow I missed that in this branch the metric is always `euclidean`. Reverted the change here. The `n_jobs` is only useful in `pairwise_distances` for other computationally expensive metrics. In fact I think we should raise a warning when `pairwise_distances` is used with an euclidean metric and `n_jobs>1`.

The benchmarks I have done above are only for `method="barnes_hut"` (default). I have added the `n_jobs` for `pairwise_distances` in `method="exact"` on general grounds as there are probably some metrics where it's helpful.

Reply:
> The benchmarks I have done above are only for method="barnes_hut" (default). I have added the n_jobs for pairwise_distances in method="exact" on general grounds as there are probably some metrics where it's helpful.

My point is that this has no impact on benchmarks above.